From 66f2323d9cafc6a27336a86045c0ff12af17fdcd Mon Sep 17 00:00:00 2001
From: Yichao Yu <yyc1992@gmail.com>
Date: Thu, 19 Oct 2023 09:02:03 -0400
Subject: [PATCH] Support loading extension in precompiled package

The global needs to exist in the base package to avoid the error

    Creating a new global in closed module `NLopt` (`Optimizer`) breaks incremental compilation because the side effects will not be permanent.
---
 src/NLopt.jl | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/NLopt.jl b/src/NLopt.jl
index df156ac..1701fde 100644
--- a/src/NLopt.jl
+++ b/src/NLopt.jl
@@ -634,6 +634,8 @@ if !isdefined(Base, :get_extension)
     include("../ext/NLoptMathOptInterfaceExt.jl")
     using .NLoptMathOptInterfaceExt
     const Optimizer = NLoptMathOptInterfaceExt.Optimizer
+else
+    global Optimizer
 end
 
 end # module
-- 
2.42.0

